src/com/cloud/chatbot/token/CommandManager.java

```diff
@@ -0,0 +1,90 @@
+package com.cloud.chatbot.token;
+
+import java.util.HashMap;
+import java.util.List;
+
+import com.cloud.chatbot.annotations.Nullable;
+import com.cloud.chatbot.exceptions.MissingInputException;
+
+
+
+/**
+ * Manages all the Tokens representing an instance of user input.
+ *
+ * The command is the head, while the details are the tail.
+ *
+ * A flag set is a flag and its sub input that have been processed. Eg "/by
+ * Tomorrow night" split into tokens is a flag set, where the flag is "/by"
+ * whose flag text is "by", and the sub input is "Tomorrow night".
+ */
+public class CommandManager extends TokenManager {
+    // Index by flag text instead of Token for faster retrieval
+    private HashMap<String, FlagManager> flagSets = new HashMap<>();
+
+    public CommandManager(String input) {
+        this.tokens = TokenManager.stringToTokens(input);
+
+        // Go through tokens to extract flag sets
+        int i = 0;
+        while (i < this.tokens.size()) {
+            Token token = this.tokens.get(i);
+
+            if (token.isFlag()) {
+                int endIndex = this.findFlagSetEnd(i);
+                List<Token> flagSet = this.removeTokens(i, endIndex);
```

Comment:
```suggestion
                List<Token> flags = this.removeTokens(i, endIndex);
```
Is it better to name this variable as flags or flagList instead of flagSet since one might be confusing about the type of this variable (Set vs List)?

Reply:
I see what you mean. I hadn't considered the potential mixup with actual `Set`s. It's actually more so referring to this definition:

https://github.com/nus-cs2103-AY2324S1/ip/blob/abb83ad37dc5a8c4d916f4b414881e9e3ed8b10d/src/com/cloud/chatbot/token/CommandManager.java#L16-L18

Since I differentiate between flags and flag sets, the `List<Token>` in question is not really a flag or flags. Instead I prefer to call it a flag set (which includes 1 flag), and is related to the method call above of `.findFlagSetEnd()`.